contracts/HolographRegistry.sol

```diff
@@ -105,21 +105,22 @@ import "./abstract/Admin.sol";
 import "./abstract/Initializable.sol";
 
 import "./interface/IHolograph.sol";
+import "./interface/IHolographRegistry.sol";
 import "./interface/IInitializable.sol";
 
 /**
  * @dev This smart contract stores the different source codes that have been prepared and can be used for bridging.
  * @dev We will store here the layer 1 for ERC721 and ERC1155 smart contracts.
  * @dev This way it can be super easy to upgrade/update the source code once, and have all smart contracts automatically updated.
  */
-contract HolographRegistry is Admin, Initializable {
+contract HolographRegistry is Admin, Initializable, IHolographRegistry {
   bytes32 constant _holographSlot = 0xb4107f746e9496e8452accc7de63d1c5e14c19f510932daa04077cd49e8bd77a;
   bytes32 constant _utilityTokenSlot = 0xbf76518d46db472b71aa7677a0908b8016f3dee568415ffa24055f9a670f9c37;
 
   /**
-   * @dev A list of smart contracts that are guaranteed secure and holographable.
+   * @dev Array of all Holographable contracts that were ever deployed on this EVM chain
```

Comment:
This is cool

Reply:
Wanted to use a bit more gas on deployment but have a way to use the blockchain to store a historical array of all holographable contract ever deployed on that chain. Used with the `getHolographableContracts` paginated function allows us to retrieve them at any given point in time.